packages/server/src/core/degree_status/postprocessing.rs

```diff
@@ -32,6 +34,54 @@ fn get_courses_of_rule_all(catalog: &Catalog) -> Vec<CourseId> {
 }
 
 impl DegreeStatus {
+    // This function returns a list of all courses with Some(grade) that belong to bank_name, sorted by grade in descending order.
+    fn get_ordered_courses_for_bank(&self, bank_name: &str) -> Vec<&CourseStatus> {
+        let mut ordered_course_statuses = self
+            .course_statuses
+            .iter()
+            .filter(|course_status| {
+                course_status.r#type == Some(bank_name.to_string()) && course_status.grade.is_some()
+            })
+            .collect::<Vec<_>>();
+
+        ordered_course_statuses.sort_by(|c1, c2| {
+            c2.grade
```

Comment:
How about:

```rust
ordered_course_statuses.sort_by(|c1, c2| {
    if let (Some(Grade::Numeric(g1)), Some(Grade::Numeric(g2))) = (&c1.grade, &c2.grade) {
        g1.partial_cmp(g2).unwrap_or(std::cmp::Ordering::Equal)
    } else {
        std::cmp::Ordering::Equal
    }
});
```
This avoids unwrapping, and therefore the comment for explaining why it can't fail is no longer needed.

Reply:
it is not correct.
if g1 is numeric and g2 is not numeric the order should be g1 > g2

I guess you meant:
ordered_course_statuses.sort_by(|c1, c2| {
    if let (Some(g1), Some(g2)) = (&c1.grade, &c2.grade) {
        g1.partial_cmp(g2).unwrap_or(std::cmp::Ordering::Equal)
    } else {
        std::cmp::Ordering::Equal
    }
});